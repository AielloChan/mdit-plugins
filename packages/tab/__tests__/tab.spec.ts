import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { tab } from "../src/index.js";

describe("tab", () => {
  const markdownIt = MarkdownIt({ linkify: true }).use(tab);

  it("should render single block", () => {
    const sources = [
      `
::: tabs
@tab test
A **bold** text.
:::
`,
      `
::: tabs

@tab test

A **bold** text.

:::
`,
    ];

    sources.forEach((item) => {
      const result = markdownIt.render(item);

      expect(result).toMatchSnapshot();
      expect(result).toContain(
        '<button type="button" class="tabs-tab-button" data-tab="0">test</button>',
      );
      expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
    });
  });

  it("should render multiple block", () => {
    const sources = [
      `
::: tabs
@tab test1
A **bold** text 1.
@tab test2
A **bold** text 2.
:::
`,
      `
::: tabs

@tab test1

A **bold** text 1.

@tab test2

A **bold** text 2.

:::
`,
    ];

    sources.forEach((item) => {
      const result = markdownIt.render(item);

      expect(result).toMatchSnapshot();
      expect(result).toContain(
        '<button type="button" class="tabs-tab-button" data-tab="0">test1</button>',
      );
      expect(result).toContain(
        '<button type="button" class="tabs-tab-button" data-tab="1">test2</button>',
      );
      expect(result).toContain("<p>A <strong>bold</strong> text 1.</p>");
      expect(result).toContain("<p>A <strong>bold</strong> text 2.</p>");
    });
  });

  describe("Should support tabs id", () => {
    it("simple id", () => {
      const source = [
        `
::: tabs#event
@tab test
A **bold** text.
:::
`,
        `
::: tabs#event

@tab test

A **bold** text.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain('data-id="event');
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0">test</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
      });
    });

    it("id with space", () => {
      const source = [
        `
::: tabs#id with space
@tab test
A **bold** text.
:::
`,
        `
::: tabs#id with space

@tab test

A **bold** text.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain('data-id="id with space');
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0">test</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
      });
    });
  });

  describe("Should support tab id", () => {
    it("simple id", () => {
      const source = [
        `
::: tabs
@tab test#id
A **bold** text.
:::
`,
        `
::: tabs

@tab test#id

A **bold** text.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain('data-id="id');
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0" data-id="id">test</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
      });
    });

    it("id with space", () => {
      const source = [
        `
::: tabs
@tab test#id with space
A **bold** text.
:::
`,
        `
::: tabs

@tab test#id with space

A **bold** text.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain('data-id="id with space');
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0" data-id="id with space">test</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
      });
    });

    it("id with multiple #", () => {
      const source = [
        `
::: tabs
@tab test#abc#def
A **bold** text.
:::
`,
        `
::: tabs

@tab test#abc#def

A **bold** text.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0" data-id="def">test#abc</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
      });
    });

    it("id with escape #", () => {
      const source = [
        `
::: tabs
@tab test\\#abc
A **bold** text.
:::
`,
        `
::: tabs

@tab test\\#abc

A **bold** text.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0">test#abc</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
      });
    });
  });

  describe("active", () => {
    it("should handle :active", () => {
      const source = [
        `
::: tabs
@tab test1
A **bold** text 1.
@tab:active test2
A **bold** text 2.
:::
`,
        `
::: tabs

@tab test1

A **bold** text 1.

@tab:active test2

A **bold** text 2.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0">test1</button>',
        );
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button active" data-tab="1" data-active>test2</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text 1.</p>");
        expect(result).toContain("<p>A <strong>bold</strong> text 2.</p>");
      });
    });

    it("should resolve first :active", () => {
      const source = [
        `
::: tabs
@tab test1
A **bold** text 1.
@tab:active test2
A **bold** text 2.
@tab:active test3
A **bold** text 3.
:::
`,
        `
::: tabs

@tab test1

A **bold** text 1.

@tab:active test2

A **bold** text 2.

@tab:active test3

A **bold** text 3.

:::
`,
      ];

      source.forEach((item) => {
        const result = markdownIt.render(item);

        expect(result).toMatchSnapshot();
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="0">test1</button>',
        );
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button active" data-tab="1" data-active>test2</button>',
        );
        expect(result).toContain(
          '<button type="button" class="tabs-tab-button" data-tab="2">test3</button>',
        );
        expect(result).toContain("<p>A <strong>bold</strong> text 1.</p>");
        expect(result).toContain("<p>A <strong>bold</strong> text 2.</p>");
        expect(result).toContain("<p>A <strong>bold</strong> text 3.</p>");
      });
    });
  });

  it("should ignore items before first @tab", () => {
    const source = [
      `
::: tabs
bala bala
@tab test\\#abc
A **bold** text.
:::
`,
      `
::: tabs

bala bala

@tab test\\#abc

A **bold** text.

:::
`,
    ];

    source.forEach((item) => {
      const result = markdownIt.render(item);

      expect(result).toMatchSnapshot();
      expect(result).toContain(
        '<button type="button" class="tabs-tab-button" data-tab="0">test#abc</button>',
      );
      expect(result).not.contain("bala bala");
      expect(result).toContain("<p>A <strong>bold</strong> text.</p>");
    });
  });

  it("should work with multiple instance", () => {
    const markdownItWithMultipleInstance = MarkdownIt({ linkify: true })
      .use(tab)
      .use(tab, {
        name: "test-tabs",
        tabsOpenRenderer: () => "<TestTabs>",
        tabsCloseRenderer: () => "</TestTabs>",
        tabOpenRenderer: () => "<TestTab>",
        tabCloseRenderer: () => "</TestTab>",
      });

    const source = [
      `
::: tabs
@tab test1
A **bold** text 1.
@tab:active test2
A **bold** text 2.
:::
::: test-tabs
@tab test3
A **bold** text 3.
@tab:active test4
A **bold** text 4.
:::
`,
      `
::: tabs

@tab test1

A **bold** text 1.

@tab:active test2

A **bold** text 2.

:::

::: test-tabs

@tab test3

A **bold** text 3.

@tab:active test4

A **bold** text 4.

:::
`,
    ];

    source.forEach((item) => {
      const result = markdownItWithMultipleInstance.render(item);

      expect(result).toMatchSnapshot();
      expect(result).toContain(
        '<button type="button" class="tabs-tab-button" data-tab="0">test1</button>',
      );
      expect(result).toContain(
        '<button type="button" class="tabs-tab-button active" data-tab="1" data-active>test2</button>',
      );
      expect(result).toContain("<p>A <strong>bold</strong> text 1.</p>");
      expect(result).toContain(
        "<TestTab><p>A <strong>bold</strong> text 3.</p>\n</TestTab>",
      );

      expect(result).toContain("<p>A <strong>bold</strong> text 2.</p>");
      expect(result).toContain(
        "<TestTab><p>A <strong>bold</strong> text 4.</p>\n</TestTab>",
      );
    });
  });
});
