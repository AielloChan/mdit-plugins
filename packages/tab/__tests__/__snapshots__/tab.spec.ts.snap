// Vitest Snapshot v1

exports[`tab > Should render multiple block 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"1\\">test2</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content\\" data-index=\\"1\\"><p>A <strong>bold</strong> text 2.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should render multiple block 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"1\\">test2</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content\\" data-index=\\"1\\"><p>A <strong>bold</strong> text 2.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should render single block 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should render single block 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > id with escape # 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test#abc</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > id with escape # 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test#abc</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > id with multiple # 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\" data-id=\\"def\\">test#abc</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\" data-id=\\"def\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > id with multiple # 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\" data-id=\\"def\\">test#abc</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\" data-id=\\"def\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > id with space 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\" data-id=\\"id with space\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\" data-id=\\"id with space\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > id with space 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\" data-id=\\"id with space\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\" data-id=\\"id with space\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > simple id 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\" data-id=\\"id\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\" data-id=\\"id\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tab id > simple id 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\" data-id=\\"id\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\" data-id=\\"id\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tabs id > id with space 1`] = `
"<div class=\\"tabs-tabs-wrapper\\" data-id=\\"id with space\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tabs id > id with space 2`] = `
"<div class=\\"tabs-tabs-wrapper\\" data-id=\\"id with space\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tabs id > simple id 1`] = `
"<div class=\\"tabs-tabs-wrapper\\" data-id=\\"event\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should support tabs id > simple id 2`] = `
"<div class=\\"tabs-tabs-wrapper\\" data-id=\\"event\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > Should work with multiple instance 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button active\\" data-tab=\\"1\\" data-active>test2</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content active\\" data-index=\\"1\\" data-active=\\"\\"><p>A <strong>bold</strong> text 2.</p>
</div>
  </div>
</div><TestTabs><TestTab><p>A <strong>bold</strong> text 3.</p>
</TestTab><TestTab><p>A <strong>bold</strong> text 4.</p>
</TestTab></TestTabs>"
`;

exports[`tab > Should work with multiple instance 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button active\\" data-tab=\\"1\\" data-active>test2</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content active\\" data-index=\\"1\\" data-active=\\"\\"><p>A <strong>bold</strong> text 2.</p>
</div>
  </div>
</div><TestTabs><TestTab><p>A <strong>bold</strong> text 3.</p>
</TestTab><TestTab><p>A <strong>bold</strong> text 4.</p>
</TestTab></TestTabs>"
`;

exports[`tab > active > Should handle :active 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button active\\" data-tab=\\"1\\" data-active>test2</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content active\\" data-index=\\"1\\" data-active=\\"\\"><p>A <strong>bold</strong> text 2.</p>
</div>
  </div>
</div>"
`;

exports[`tab > active > Should handle :active 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button active\\" data-tab=\\"1\\" data-active>test2</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content active\\" data-index=\\"1\\" data-active=\\"\\"><p>A <strong>bold</strong> text 2.</p>
</div>
  </div>
</div>"
`;

exports[`tab > active > Should resolve first :active 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button active\\" data-tab=\\"1\\" data-active>test2</button>
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"2\\">test3</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content active\\" data-index=\\"1\\" data-active=\\"\\"><p>A <strong>bold</strong> text 2.</p>
</div><div class=\\"tabs-tab-content\\" data-index=\\"2\\"><p>A <strong>bold</strong> text 3.</p>
</div>
  </div>
</div>"
`;

exports[`tab > active > Should resolve first :active 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test1</button>
    <button type=\\"button\\" class=\\"tabs-tab-button active\\" data-tab=\\"1\\" data-active>test2</button>
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"2\\">test3</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text 1.</p>
</div><div class=\\"tabs-tab-content active\\" data-index=\\"1\\" data-active=\\"\\"><p>A <strong>bold</strong> text 2.</p>
</div><div class=\\"tabs-tab-content\\" data-index=\\"2\\"><p>A <strong>bold</strong> text 3.</p>
</div>
  </div>
</div>"
`;

exports[`tab > should ignore items before first @tab 1`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test#abc</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;

exports[`tab > should ignore items before first @tab 2`] = `
"<div class=\\"tabs-tabs-wrapper\\">
  <div class=\\"tabs-tabs-header\\">
    <button type=\\"button\\" class=\\"tabs-tab-button\\" data-tab=\\"0\\">test#abc</button>
  </div>
  <div class=\\"tabs-tabs-container\\">
<div class=\\"tabs-tab-content\\" data-index=\\"0\\"><p>A <strong>bold</strong> text.</p>
</div>
  </div>
</div>"
`;
